### PR TITLE
chore: 🧑‍💻 switch from flake8 to ruff for linting

### PR DESCRIPTION
**Pull request by zoido**
`flake8` with plugins and `isort` was replaced by `ruff`.
`ruff`'s `isort` does not support profiles, so I went with config that close to `appnexus` profile.

### CODE REVIEW

File: src/h2o_discovery/__init__.py [hunk 1]
```diff
@@ -108,6 +108,6 @@ def _lookup_and_load(
             " Alternatively, you can create configure your local environment with"
             " the with H2O CLI with and/or use H2OCONFIG environment variable"
             " (see https://docs.h2o.ai/h2o-ai-cloud/developerguide/cli)."
-        )
+        ) from None
```
Comment (johnygomez) [inline src/h2o_discovery/__init__.py:111]: QQ: What is the reason to suppress the exception context here?

**Review by johnygomez — APPROVED** [review]
LGTM, I came across this tool some time ago, but seemed to me it wasn't yet very stable for production. Don't you know why they still keep it in the pre-alpha version? Otherwise it looks very good

File: pyproject.toml [hunk 2]
```diff
@@ -67,40 +67,36 @@ template = "test"
...
   "check",
 ]
 
-[tool.hatch.envs.docs]
-dependencies = [
-  "pydoc-markdown==4.6.4",
-]
-
-[tool.hatch.envs.docs.scripts]
-generate = "pydoc-markdown"
```
[the remark this answers is not in the document]
Comment (zoido) [reply pyproject.toml:98]: OMG! Thanks. Error. This should not have been removed.

FIXED!

File: src/h2o_discovery/__init__.py (hunk 1 above)
Comment (zoido) [reply src/h2o_discovery/__init__.py:111]: This change is just addressing linting error.

```
src/h2o_discovery/__init__.py:104:9: B904 Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling
```

https://docs.astral.sh/ruff/rules/raise-without-from-inside-except/

So no extra suppressing as the whole chain is custom and explicitly raised.

**Comment by zoido** [comment]
> Don't you know why they still keep it in the pre-alpha version? Otherwise it looks very good

My guess is that it's being actively developed.

**Pull request merged**